### PR TITLE
Fixes #901 : Added documentation for "step" in dti

### LINKED ISSUE #901: DTI `step` parameter not documented.
Need to document what this does at the TensorModel level.

### PR DESCRIPTION
**Pull request by sahmed95**
Fixes #901

### CODE REVIEW

**Comment by arokem** [comment]
Do you want to sort out those PEP8 issues, while we're here?

File: dipy/reconst/dti.py [hunk 1]
```diff
@@ -1132,6 +1132,9 @@ def predict(self, gtab, S0=1, step=None):
            The mean non-diffusion weighted signal in each voxel. Default: 1 in
            all voxels.
 
+        step : int
+            The chunk size as a number of voxels.
```
Comment (arokem) [inline dipy/reconst/dti.py]: We might want to add a bit more explanation here (after the first sentence). Something like: "to increase speed of processing, tensor fitting is done simultaneously over many voxels. This parameter sets the number of voxels that will be fit at once in each iteration. Note that a larger number here should speed things up, but should also take up more memory, so keep an eye on your computers memory consumption as you increase this number.

Also, we should mention that this is an optional parameter, and that it has a default value (10,000).

We should also add all this to the documentation of the `TensorModel` object.

Comment (sahmed95) [inline dipy/reconst/dti.py]: Ok. I will add this and also update the documentaion for TensorModel object. For the pep8 issues, I can do it in this commit itself. But, if making a new branch just for pep8 isn't a big deal, I guess its better to create a seperate issue and fix pep8 there.

Comment (sahmed95) [inline dipy/reconst/dti.py]: Should the explanation come directly under the definition of step or I should include it under Notes. Also, the line - the increase in step value "should" speed things up but it ~~should~~ will also take up more memory - Is this correct? Regarding the memory 
```python
    """
            The chunk size as a number of voxels. Optional parameter with default value 10,000.

            In order to increase speed of processing, tensor fitting is done simultaneously
            over many voxels. This parameter sets the number of voxels that will be fit at 
            once in each iteration. A larger step value should speed things up, but will 
            also take up more memory. It is advisable to keep an eye on memory consumption 
            as this value is increased.
    """
```

**Comment by sahmed95** [comment]
@arokem Is this fine?

**Comment by arokem** [comment]
I was hoping to also document this at the level of the TensorModel object. Maybe in the `__init__`? It's specific to only some fitting methods (ols, wls), but these are rather important, and we want to give even the users who don't want to dig some information about this, I think.

**Comment by sahmed95** [comment]
So all the functions [here](https://github.com/nipy/dipy/blob/master/dipy/reconst/dti.py#L1122)

**Comment by arokem** [comment]
Rather, somewhere here: https://github.com/nipy/dipy/blob/master/dipy/reconst/dti.py#L696

**Comment by sahmed95** [comment]
Done :smile:

File: dipy/reconst/dti.py [hunk 2]
```diff
@@ -721,6 +722,17 @@ def __init__(self, gtab, fit_method="WLS", *args, **kwargs):
...
+        Note
+        -----
+        In order to increase speed of processing, tensor fitting is done simultaneously
+        over many voxels. Many fit_methods use the 'step' parameter to set the number of
+        voxels that will be fit at once in each iteration. This is the chunk size as a 
+        number of voxels. A larger step value should speed things up, but it will also 
+        take up more memory. It is advisable to keep an eye on memory consumption as 
+        this value is increased.
+
+        Example : In iter_fit_tensor() we have a default step value of 1e4
```
Comment (arokem) [inline dipy/reconst/dti.py]: Please change 
`iter_fit_tensor()`=> `:func:`iter_fit_tensor``

**Comment by arokem** [comment]
Cool. One more small change, so that this renders fine in sphinx (and add's a link in the html to the right function). Thanks for picking up those PEP8 bits as well!

**Comment by sahmed95** [comment]
Done.

**Comment by arokem** [comment]
One more small thing - in the sphinx markup, the name of the function has
to be surrounded by backticks. See here:
http://stackoverflow.com/questions/21289806/link-to-class-method-in-python-docstring

**Comment by sahmed95** [comment]
Updated the backticks

**Comment by arokem** [comment]
This now looks fine to me. Anyone else want to give this a look?

**Pull request merged**